### PR TITLE
Accessibility remedial

### PR DESCRIPTION
**Pull request by kalpaitch**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -25,14 +25,16 @@
         }
     ],
     "require": {
-        "php": "~8.1.0",
+        "php": "~8.1.2",
```
Comment (kalpaitch) [inline composer.json:28]: @robwildolivejar It was actually this change to the compose.json file.

File: composer.json [hunk 2]
```diff
@@ -32,7 +32,9 @@
         "composer/installers": "^1.9",
         "cweagans/composer-patches": "^1.7",
         "deblan/csv-validator": "dev-master",
+        "drupal/a11y_autocomplete_element": "^1.0",
         "drupal/address": "~1.11.0",
+        "drupal/admin_toolbar": "^3.4",
```
Comment (kalpaitch) [inline composer.json:37]: Not sure this is strictly necessary, we don't really have admin users (aside from us devs)??

File: composer.json [hunk 3]
```diff
@@ -179,7 +181,9 @@
                 "Form API #ajax not working on link element": "https://www.drupal.org/files/issues/2023-01-11/core-allow-form-ajax-for-links-2915954-9.5.1.patch",
                 "Add events for matching entity hooks": "https://www.drupal.org/files/issues/2022-12-22/2551893-99.patch",
                 "Non-exposed and exposed default sort order": "https://www.drupal.org/files/issues/2021-02-16/3120847-20.patch",
-                "PostgreSQL: email case insensitivity": "https://www.drupal.org/files/issues/2022-12-19/drupal-case_sensitive_email-2490294-58-d95.patch"
+                "PostgreSQL: email case insensitivity": "https://www.drupal.org/files/issues/2022-12-19/drupal-case_sensitive_email-2490294-58-d95.patch",
+                "Add aria labelling for form states": "./patches/states-aria.patch",
```
Comment (kalpaitch) [inline composer.json:185]: Let's try and create a drupal issue for each of these patches and link to this. That way we will have a record of why the patch has been applied and what it does.

File: package.json [hunk 4]
```diff
@@ -22,7 +21,8 @@
   },
   "scripts": {
     "gulp": "gulp",
-    "install-govuk-theme": "cd web/themes/contrib/govuk_theme && npm install && npm run gulp build -- --ce --ct --cf --cl",
-    "install-par-theme": "cd web/themes/custom/govuk_par && npm install"
+    "install-govuk-par": "cd web/themes/custom/govuk_par && npm install && npm run gulp",
```
Comment (kalpaitch) [inline package.json:24]: Will review this module separately to see why contrib module doesn't meet the requirements here. Could always try and add you as a maintainer of this...

File: patches/inline_form_errors.patch [hunk 5]
```diff
@@ -0,0 +1,218 @@
+diff --git a/core/lib/Drupal/Core/Form/FormElementHelper.php b/core/lib/Drupal/Core/Form/FormElementHelper.php
```
Comment (kalpaitch) [inline patches/inline_form_errors.patch:1]: I feel like I've seen a drupal.org issue for this, could you share if there is anything this is based on.

File: sync/block.block.analyticscookies_2.yml [hunk 6]
```diff
@@ -0,0 +1,25 @@
+uuid: dd547e01-7e4c-40fb-ae49-3f00110d02f6
```
Comment (kalpaitch) [inline sync/block.block.analyticscookies_2.yml:1]: Why is there a second analytics cookies block?

I guess one for each of the themes, just trying to work out why this has changed when the contrib govuk_theme didn't have one. I'm not sure how these two themes link together at this point, will have a bit more of a dig, any light you can shed on this?

File: web/modules/custom/par_flows/src/ParRedirectTrait.php [hunk 7]
```diff
@@ -75,7 +75,7 @@ public function getLinkByRoute($route, $route_params = [], $link_options = [], $
   public function getLinkByUrl(Url $url, $text = '', $link_options = []) {
     $link_options += [
       'absolute' => TRUE,
-      'attributes' => ['class' => 'flow-link']
+      'attributes' => ['class' => 'govuk-link']
```
Comment (kalpaitch) [inline web/modules/custom/par_flows/src/ParRedirectTrait.php:78]: Completely agree with adding the class, but lets keep the other class there for now, no need to introduce more breaking changes than necessary.

File: web/modules/custom/par_forms/src/Plugin/ParForm/ParBusinessSizeForm.php [hunk 8]
```diff
@@ -42,7 +42,7 @@ public function getElements($form = [], $cardinality = 1) {
 
     $form['info'] = [
       '#markup' => t('Enter the number of associations in your membership list'),
```
Comment (kalpaitch) [inline web/modules/custom/par_forms/src/Plugin/ParForm/ParBusinessSizeForm.php:44]: If we're making this change we could probably replace it with the more appropriate render element:
```
          '#type' => 'html_tag',
          '#tag' => 'h2',
          '#attributes' => ['class' => ['govuk-heading-m']],
          '#value' => $this->t('Enter the number of associations in your membership list'),
```

File: web/modules/custom/par_forms/src/Plugin/ParForm/ParContactDisplay.php [hunk 9]
```diff
@@ -98,12 +98,12 @@ public function loadData($cardinality = 1) {
...
       $form['message_intro'] = [
-        '#type' => 'fieldset',
-        'title' => [
+        '#type' => 'container',
+        'heading' => [
           '#type' => 'html_tag',
           '#tag' => 'h2',
           '#value' => $this->t('Contacts'),
-          '#attributes' => ['class' => ['heading-large']],
+          '#attributes' => ['class' => ['z']],
```
Comment (kalpaitch) [inline web/modules/custom/par_forms/src/Plugin/ParForm/ParContactDisplay.php:106]: Little oopsie here, think the class should be `govuk-heading-l`

File: web/modules/custom/par_forms/src/Plugin/ParForm/ParContactDisplay.php [hunk 10]
```diff
@@ -143,7 +143,7 @@ public function getElements($form = [], $cardinality = 1) {
       }
 
       $form['contact'] = [
-        '#type' => 'fieldset',
+        '#type' => 'container',
         '#weight' => 1,
         '#attributes' => ['class' => ['grid-row', 'form-group', 'contact-details']],
```
Comment (kalpaitch) [inline web/modules/custom/par_forms/src/Plugin/ParForm/ParContactDisplay.php:148]: I can see a `grid-row` class here that should probably be updated to `govuk-grid-row` as per the other changes. Maybe just one final check that all these classes have been changed??

File: web/modules/custom/par_forms/src/Plugin/ParForm/ParGdprForm.php [hunk 11]
```diff
@@ -64,6 +64,7 @@ public function getElements($form = [], $cardinality = 1) {
 
     $form['summary'] = [
       '#theme' => 'item_list',
+      '#list_header_tag' => 'h2',
```
Comment (kalpaitch) [inline web/modules/custom/par_forms/src/Plugin/ParForm/ParGdprForm.php:67]: Looks great. I was wondering if this applies to any other elements (particularly those where there is just one question on a page and the label should become the primary heading) ?? Probably a separate conversation though.

File: web/modules/custom/par_notification/src/Controller/ParAccessDeniedController.php [hunk 12]
```diff
@@ -63,11 +63,11 @@ public function build(Request $request) {
       && !user_load_by_mail($message->get('field_to')->getString()));
 
     // Column size is determined by whether invitations should be shown.
-    $column_size = $allow_invitations ? 'column-one-half' : 'column-full';
+    $column_size = $allow_invitations ? 'column-one-half' : 'govuk-grid-column-full';
```
Comment (kalpaitch) [inline web/modules/custom/par_notification/src/Controller/ParAccessDeniedController.php:66]: Need to change the `column-one-half` class as well...

File: web/modules/features/par_enforcement_raise_flows/src/Form/ParChecklistForm.php [hunk 13]
```diff
@@ -89,7 +89,7 @@ public function buildForm(array $form, FormStateInterface $form_state) {
         '#title' => $this->t('Discuss this enforcement'),
         '#url' => Url::fromRoute('par_enquiry_send_flows.select_authority', $this->getRouteParams()),
         '#prefix' => '<p class="form-group">',
-        '#suffix' => '<br><br><br></p>',
+        '#suffix' => '</p>',
```
Comment (kalpaitch) [inline web/modules/features/par_enforcement_raise_flows/src/Form/ParChecklistForm.php:92]: Just a reminder to check spacing on this element.

File: web/modules/features/par_partnership_flows/src/Form/ParPartnershipFlowsSicCodeForm.php [hunk 14]
```diff
@@ -59,7 +59,7 @@ public function buildForm(array $form, FormStateInterface $form_state, ParDataPa
     }
 
     $form['sic_code'] = [
-      '#type' => 'select',
+      '#type' => 'a11y_autocomplete',
```
Comment (kalpaitch) [inline web/modules/features/par_partnership_flows/src/Form/ParPartnershipFlowsSicCodeForm.php:62]: I see this element change is causing a bit of trouble, will review this separately as well. There is a lot of work to do on the sic code selection, and I don't think this element is necessarily the solution going forward, may be better just to create a separate ticket for this work??

File: web/modules/custom/par_notification/src/Controller/ParAccessDeniedController.php (hunk 12 above)
Comment (kalpaitch) [reply web/modules/custom/par_notification/src/Controller/ParAccessDeniedController.php:66]: Could be useful to create a linting plugin/script to check we've replaced all the old class names (if it's a simple task).

File: composer.json (hunk 2 above)
Comment (robwildolivejar) [reply composer.json:37]: Not strictly necessary but super useful - I enable it for every task I pick up

File: composer.json (hunk 3 above)
Comment (robwildolivejar) [reply composer.json:183]: The postgreSQL is already there, the aria-labels one I will do

File: package.json (hunk 4 above)
Comment (robwildolivejar) [reply package.json:24]: Its been a long time since any work was done for this

File: sync/block.block.analyticscookies_2.yml (hunk 6 above)
Comment (robwildolivejar) [reply sync/block.block.analyticscookies_2.yml:1]: One for each enabled theme

File: web/modules/custom/par_flows/src/ParRedirectTrait.php (hunk 7 above)
Comment (robwildolivejar) [reply web/modules/custom/par_flows/src/ParRedirectTrait.php:78]: No need to keep legacy classes, the testers have updated the tests accordingly

File: web/modules/custom/par_forms/src/Plugin/ParForm/ParContactDisplay.php (hunk 9 above)
Comment (robwildolivejar) [reply web/modules/custom/par_forms/src/Plugin/ParForm/ParContactDisplay.php]: Good spot, have fixed this

File: web/modules/custom/par_forms/src/Plugin/ParForm/ParContactDisplay.php (hunk 10 above)
Comment (robwildolivejar) [reply web/modules/custom/par_forms/src/Plugin/ParForm/ParContactDisplay.php]: Good spot, I have changed this and searched the rpo for other occurrences

File: web/modules/features/par_enforcement_raise_flows/src/Form/ParChecklistForm.php (hunk 13 above)
Comment (robwildolivejar) [reply web/modules/features/par_enforcement_raise_flows/src/Form/ParChecklistForm.php:92]: The artificial padding this adds is now picked up via CSS and applied correctly, albeit on the form-group elements in a one of the 4 types of row styles

File: web/modules/features/par_partnership_flows/src/Form/ParPartnershipFlowsSicCodeForm.php (hunk 14 above)
Comment (robwildolivejar) [reply web/modules/features/par_partnership_flows/src/Form/ParPartnershipFlowsSicCodeForm.php:62]: the SIC code is now back as a select

**Review by robwildolivejar — COMMENTED** [review]
Review supdated

**Review by robwildolivejar — COMMENTED** [review]
Updated merge regression

**Pull request merged**